modules/admin/src/main/java/org/motechproject/admin/internal/service/impl/ModuleAdminServiceImpl.java

```diff
@@ -263,6 +278,66 @@ public ExtendedBundleInformation getBundleDetails(long bundleId) {
         }
     }
 
+    private String parseDependencyVersion(Dependency dependency, MavenRepositorySystemSession mvnRepository, RepositorySystem system,
+                                          RemoteRepository remoteRepository, PomInformation pomInformation) throws ArtifactResolutionException {
+        String parsedVersion;
+        String version = dependency.getArtifact().getVersion();
+
+        if (StringUtils.isEmpty(version)) {
+            parsedVersion = "[0,)";
+        } else if (version.contains("${")) {
+            String propertyName = StringUtils.remove(version, "${");
+            propertyName = StringUtils.remove(propertyName, '}');
+
+            parsedVersion = getVersionFromProperties(propertyName, mvnRepository, system, remoteRepository, pomInformation);
+            if (parsedVersion == null) {
+                throw new MotechException(String.format("The property %s used in dependency %s cannot be found in pom " +
```

Comment:
Is it wise for use to throw an exception, given that our parsing seems limited? Perhaps just log an error and try to proceed anyway?